gpu_memory_management/cupy_pybind.cu

```diff
@@ -39,6 +39,26 @@ PYBIND11_MODULE(gpuMemManagement, m)
   m.def("test_wrong_dtype_float", &test_wrong_dtype_float);
   m.def("test_wrong_dtype_int", &test_wrong_dtype_int);
   m.def("test_wrong_dtype_complex", &test_wrong_dtype_complex);
+
+  //missmatch error always appeared based on the order of types
+  //ex : if cupy with double type is used, them there will be 3 missmatch error for uint16, uint32, and float
+  //ex : if cupy with complex128 type is used, them there will be 5 missmatch error for uint16, uint32, float, double, and complex64
+  //this error appeard with "pytest -s"
+  m.def("test_custom_cupy_template_function", &test_custom_cupy_template_function<uint16_t>);
+  m.def("test_custom_cupy_template_function", &test_custom_cupy_template_function<uint32_t>);
+  m.def("test_custom_cupy_template_function", &test_custom_cupy_template_function<float>);
+  m.def("test_custom_cupy_template_function", &test_custom_cupy_template_function<double>);
+  m.def("test_custom_cupy_template_function", &test_custom_cupy_template_function<complex<float>>);
+  m.def("test_custom_cupy_template_function", &test_custom_cupy_template_function<complex<double>>);
```

Comment:
Do you know what happens, if you overload a python function. Normally, it is forbidden. I don't know, what pybind11 is doing.

Reply:
I do it in the previous phase retrieval project and it is working fine

Reply:
But in the last project, you didn't used custom caster and I'm not sure, if you also didn't used templates. You should test it without overloading. It is a cheep try.

Reply:
I already tried without overloading before this commit. Without overloading, I get a pybind11 error. I look into it and I found that I need an overloading